### PR TITLE
fix(chromadb): queries without embeddings result in program crash

### PR DESCRIPTION
**Pull request by poxrud**
# Description

When using `VectorStoreQuery` to query _chromadb_ by metadata only, without an embedding, it results in 
a program crash with the following error message: 

```ValueError: You must provide one of query_embeddings, query_texts, query_images, or query_uris.```

# Why is this a problem?

When using the `VectorIndexAutoRetriever`, it will sometimes suggest a metadata only search, without an embedding, 
and this will cause a program crash with the above error message.

This fix is to use the chromadb collection's `get()` method instead of `query()` for cases where a metadata only search is required.

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Added a new test ```test_add_to_chromadb_and_query_by_metafilters_only``` into `test_chromadb.py`.
Run it with:

```sh
pytest llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/tests/test_chromadb.py
```

- [x] Added new unit/integration tests
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
I think the idea is to provide an "empty vector" for the auto retriever to query with. But this works too

File: llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py [hunk 1]
```diff
@@ -291,13 +291,29 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
         else:
             where = kwargs.pop("where", {})
 
-        results = self._collection.query(
+        if not query.query_embedding:
+            return self._get(limit=query.similarity_top_k, where=where, **kwargs)
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py:295]: Should there still be a limit if its purely a filter query?

File: llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py [hunk 2]
```diff
@@ -291,13 +291,29 @@ def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResul
...
+    ) -> VectorStoreQueryResult:
+        results = self._collection.query(
+            query_embeddings=query_embeddings,
+            n_results=n_results,
+            where=where,
+            **kwargs,
+        )
+
+        print(f"QUERY RES: {results}")
+        print(f"EMBEDDINGS: {query_embeddings}")
```
Comment (logan-markewich) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py:315]: remove debug prints

**Comment by poxrud** [comment]
> I think the idea is to provide an "empty vector" for the auto retriever to query with. But this works too

As I understand it, the "empty vector" is used when the query string is empty and is more of a default search string in the form of an embedding. It doesn't allow for a metadata only searches.

File: llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py (hunk 1 above)
Comment (poxrud) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-chroma/llama_index/vector_stores/chroma/base.py:295]: I think so, from the perspective of the end-user they don't know whether the query() or get() method gets used, so when they set `smililarity_top_k` they expect a max of this number of results.

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**